otherlibs/systhreads/thread.mli

```diff
@@ -41,8 +42,31 @@ val id : t -> int
    is an integer that identifies uniquely the thread.
    It can be used to build data structures indexed by threads. *)
 
+exception Exit
+(** Exception raised by user code to initiate termination of the
+    current thread.
+    In a thread created by [{!Thread.create} funct arg], if the
+    {!Thread.Exit} exception reaches the top of the application
+    [funct arg], it has the effect of terminating the current thread
+    silently.  In other contexts, there is no implicit handling of the
+    {!Thread.Exit} exception. *)
+
 val exit : unit -> unit
-(** Terminate prematurely the currently executing thread. *)
+[@@ocaml.deprecated "Use 'raise Thread.Exit' instead."]
+(** Raise the {!Thread.Exit} exception.
+    In a thread created by {!Thread.create}, this will cause the thread
+    to terminate prematurely, unless the thread function handles the
+    exception itself.  {!Fun.protect} finalizers and catch-all
+    exception handlers will be executed.
+
+    To make it clear that an exception is raised and will trigger
+    finalizers and catch-all exception handlers, it is recommended
+    to write [raise {!Thread.Exit}] instead of [{!Thread.exit} ()].
```

Comment:
I do not find this explanation for writing directly `raise Thread.Exit` convincing. You can just say that the previous behaviour was incompatible which justifies deprecating, and point out that converting to the raising form might need to review and adapt the program. In particular it might be good to point out that it is no longer guaranteed that a call to Thread.exit terminates the thread (I find the wording "unless the thread function handles the exception itself" unclear about accidental "handlings").